### ISSUE TITLE
Enums don't require a type definition

### ISSUE THREAD

**Issue by magopian** [mozilla-services/react-jsonschema-form#66]
(splitted out from #25)

Example in http://json-schema.org/example2.html shows an `enum` field doesn't seem to require a `type` property defined:

``` js
"fstype": {
  "enum": ["ext3", "ext4", "btrfs"]
},
```

**Comment by leplatrem** [comment]
In that case you assume `"string"`, but should we introspect and guess the type from enum values?

**Comment by n1k0** [comment]
> should we introspect and guess the type from enum values?

And how about 

```js
"fstype": {
  "enum": ["str", 2, {foo: [1, 2]}]
},
```

Oh my.

**Comment by n1k0** [comment]
> http://json-schema.org/latest/json-schema-validation.html#anchor76
> The value of this keyword MUST be an array. This array MUST have at least one element. 
> Elements in the array MUST be unique.
>
> Elements in the array MAY be of any type, including null.

**Comment by epicfaace** [comment]
This was fixed by #668
